### DISCUSSION TITLE
Edit command/file picker/command pallette input with normal mode

### DISCUSSION THREAD

**Discussion by sburris0** [helix-editor/helix#10059] [Ideas]
There does not appear to be a way to edit commands (e.g. `:buffer-close`) using normal helix movement keys, users are stuck with arrow keys. I expect this to be an increasingly frustrating issue as plugin support is completed, which I assume will allow plugins to add custom commands that are more complicated than those included in helix. This issue is also present in other dialogs such as file search, command palette, etc. Being able to use normal mode in dialogs gives users a way to stick to normal movement commands vs `ctrl-n` and `ctrl-p`.

This previously bothered me in vim. It turns out that vim does have a solution for this: `ctrl+f`.
This hotkey opens a new pane in which the command being typed may be edited like any other text.
Demonstration of this: https://youtu.be/pmQMTfSABhw?t=901

Opening a new buffer like vim is one solution.
I prefer a way to keep the cursor in the same place because it can be used more universally: command input, file search input, command pallette, etc.

One hurdle is that normally normal mode is entered with `esc`, which closes whatever dialog the user is looking at. I think that using `ctrl+esc` to enter normal mode here is a reasonable solution, with `i` to return to insert mode as usual.

Example of use:
1. Accidentally type `:uffer-close`
2. Hit `ctrl+esc` to enter normal mode
3. `gs` to go to start of line
4. Re-enter insert mode with `i`
5. Correct command be inserting missing `b`

**Comment by haxfn** [comment]
C-f would be great for command history editing. For now I am using the helix's emacs like binds in the "command mode line" and in other places like the file finder etc (https://docs.helix-editor.com/keymap.html#insert-mode)
